__tests__/app.test.js

```diff
@@ -365,3 +367,125 @@ describe("PATCH /api/articles/:article_id", () => {
+    .expect(200)
+    .then((response) => {
+      let articles = response.body
+      expect(articles.length).toBe(1)
+      articles.forEach((article) => {
+        expect(article).toHaveProperty("article_id", expect.any(Number));
+        expect(article).toHaveProperty("title", expect.any(String));
+        expect(article).toHaveProperty("topic", expect.any(String));
+        expect(article).toHaveProperty("author", expect.any(String));
+        expect(article).toHaveProperty("created_at", expect.any(String));
+        expect(article).toHaveProperty("votes", expect.any(Number));
+        expect(article).toHaveProperty("article_img_url", expect.any(String));
+        expect(article).toHaveProperty("comment_count", expect.any(String));
+      });
+  })
+  })
+  test('400: topic provided does not exist', () => {
+    return request(app)
+      .get("/api/articles?topic=dogs")
+    .expect(400)
+    .then((response) => {
+      expect(response.body.msg).toBe('Bad request')
+  })
+  })
+})
+describe("GET /api/articles?sort_by=:property", () => {
+  test("200: sort_by query provided, sorts by property, returns articles", () => {
+    return request(app)
+      .get("/api/articles?sort_by=votes")
+    .expect(200)
+      .then((response) => {
+        let articles = response.body
+        expect(articles).toBeSortedBy('votes', { descending: true })
+        expect(articles.length).toBe(13)
+        articles.forEach((article) => {
+          expect(article).toHaveProperty("article_id", expect.any(Number));
+          expect(article).toHaveProperty("title", expect.any(String));
+          expect(article).toHaveProperty("topic", expect.any(String));
+          expect(article).toHaveProperty("author", expect.any(String));
+          expect(article).toHaveProperty("created_at", expect.any(String));
+          expect(article).toHaveProperty("votes", expect.any(Number));
+          expect(article).toHaveProperty("article_img_url", expect.any(String));
+          expect(article).toHaveProperty("comment_count", expect.any(String));
+        });
+      
+  })
+  })
+  test('200: if no sort_by query provided, defaults to created_at', () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((response) =>{
+        let articles = response.body
+        expect(articles).toBeSortedBy('created_at', { descending: true })
+    })
+  })
+
+  test('400: property provided does not exist', () => {
+    return request(app)
+      .get("/api/articles?sort_by=dogs")
+    .expect(400)
+    .then((response) => {
+      expect(response.body.msg).toBe('Bad request')
+  })
+  })
+
+})
+
+describe("GET /api/articles?order=:order", () => {
+  test('200: sets order to asc or desc', () => {
+    return request(app)
+      .get("/api/articles?order=asc")
+    .expect(200)
+      .then((response) => {
+        let articles = response.body
+        expect(articles).toBeSortedBy('created_at', { ascending: true })
```

Comment:
I don't think ascending is an option, I think this only works because descending defaults to false